drop-boxes/register-omero-metadata/image_registration_process.py

```diff
@@ -89,43 +72,115 @@ def requestOmeroDatasetId(self, project_code=None, sample_code=None):
+            break
 
         return ds_id
 
     def registerImageFileInOmero(self, file_path, dataset_id):
 
         cmd_list = list(self._init_cmd_list)
-        cmd_list.append( "python backendinterface.py -f " + file_path + " -d " + str(dataset_id) )
+        cmd_list.append( "omero-bifrost push img-file " + file_path + " " + str(dataset_id) + " --to-xml" )
 
-        commands = ""
-        for cmd in cmd_list:
-            commands = commands + cmd + "\n"
+        out, err = self.executeCommandList(cmd_list)
 
-        process = Popen( "/bin/bash", shell=False, universal_newlines=True, stdin=PIPE, stdout=PIPE, stderr=PIPE )
-        out, err = process.communicate( commands )
+        id_list = []
+        output_xml = ET.fromstring(str(out))
+        for output_item in output_xml:
+            img_id = output_item.attrib["id"]
+            if img_id.isdigit():
+                id_list.append(img_id)
+            else:
+                return []
 
-        id_list = str(out).split()
-        for img_id in id_list:
-            if not img_id.isdigit():
+        return id_list
+
+    def standardImageFolderRegistration(self, image_folder_path, dataset_id):
+        """Imports folder using the standard process of omero-py
+        """
+
+        cmd_list = list(self._init_cmd_list)
+        cmd_list.append( "omero-bifrost push img-folder " + image_folder_path + " " + str(dataset_id) + " --to-xml" )
+
+        out, err = self.executeCommandList(cmd_list)
+
+        id_list = []
+        output_xml = ET.fromstring(str(out))
+        for output_item in output_xml:
+            img_id = output_item.attrib["id"]
+            if img_id.isdigit():
+                id_list.append(img_id)
+            else:
                 return []
 
         return id_list
+
+    def registerImageFolder(self, image_folder_path, dataset_id):
+
+        return self.standardImageFolderRegistration(image_folder_path, dataset_id)
+
+    def attachFileToImage(self, file_path, image_id):
+
+        cmd_list = list(self._init_cmd_list)
+
+        cmd_list.append( "omero-bifrost push file-atch " + file_path + " " + str(image_id) )
+
+        out, err = self.executeCommandList(cmd_list)
+
+        return 0
+
+    def tagOmeroImage(self, image_id, tag_value):
+
+        cmd_list = list(self._init_cmd_list)
+
+        cmd_list.append( "omero-bifrost push img-tag " + str(image_id) + " " + tag_value )
+
+        out, err = self.executeCommandList(cmd_list)
+
+        return 0
+
+    def registerOmeroKeyValuePairs(self, image_id, property_map):
+        """Registers the property map as key-value pairs in the OMERO server.
+        """
         
+        cmd_list = list(self._init_cmd_list)
+
+        key_value_str = ""
+        for key in property_map.keys(): 
+            key_value_str = key_value_str + "--kv-pair " + str(key) + ":" + str(property_map[key]) + " "
+        key_value_str = key_value_str[:len(key_value_str)-2] #remove last two chars
```

Comment:
What's the purpose of removing the last two chars here? I think a small line explaining your logic would help 👍 
```suggestion
        for key in property_map.keys(): 
            key_value_str = key_value_str + "--kv-pair " + str(key) + ":" + str(property_map[key]) + " "
        key_value_str = key_value_str[:len(key_value_str)-2] #remove last two chars
```

Reply:
Good point, thank you. The idea was to remove the last white space, I added a comment. But you also found a bug :) its fixed now

Reply:
To remove whitespaces i'd recommend to use the python native [strip method](https://stackoverflow.com/questions/1185524/how-do-i-trim-whitespace)